packages/babel-core/src/config/validation/option-assertions.js

```diff
@@ -337,7 +337,7 @@ function assertPluginItem(loc: GeneralPath, value: mixed): PluginItem {
       if (
         opts !== undefined &&
         opts !== false &&
-        (typeof opts !== "object" || Array.isArray(opts))
+        (typeof opts !== "object" || Array.isArray(opts) || !opts)
```

Comment:
I would prefer to be explicit here, to avoid confusion with the `false` and `undefined` cases.
```suggestion
        (typeof opts !== "object" || Array.isArray(opts) || opts === null)
```